server/src/test/java/org/elasticsearch/index/reindex/ReindexRequestTests.java

```diff
@@ -41,11 +48,97 @@
  */
 public class ReindexRequestTests extends AbstractBulkByScrollRequestTestCase<ReindexRequest> {
 
+    private final String matchAll = "{\"match_all\":{}}";
```

Comment:
I would prefer `new BytesArray("{ \"foo\" : \"bar\" }"`, makes it more explicit what we rely on (JSON and not query-validity).